crates/autopilot/src/decoded_settlement.rs

```diff
@@ -209,12 +220,11 @@ impl DecodedSettlement {
             .context("decoding tokenized settlement calldata failed")?;
 
         let (tokens, clearing_prices, trades, interactions) = decoded;
-        Ok(Self {
-            tokens,
-            clearing_prices,
-            trades: trades
-                .into_iter()
-                .map(|trade| DecodedTrade {
+        let trades = trades
+            .into_iter()
+            .filter_map(|trade| {
+                let mut trade = DecodedTrade {
+                    order_uid: Default::default(),
```

Comment:
This is weird. Is it valid to instantiate a DecodedTrade with such an invalid `order_uid` (in this case the field should be optional to indicate "this is not a real id")? If not, I think we should give it a constructor that can take the domain_seperator and tokens and returns an error if the uid cannot be recovered instead of allowing callsites to create invalid representations.

Reply:
I am always against such mutations, but this requires much more refactoring, and Dusan mentioned it will be changed in the upcoming PRs. I can make it optional and return an error in case the order is None when it is required.

Reply:
Maybe I don't understand the issue. I'm in favor of keeping this field, just enforcing the struct gets instantiated via a constructor and if it fails we are not returning a broken object but an error. But sue, it doesn't make a huge difference so fee free to ignore if it causes issues.

Reply:
A proposal: https://github.com/cowprotocol/services/pull/2688/commits/9cac26491f03d972d82ec00dd7c12f3edb6c0198